daemon/algod/api/server/v2/handlers.go

```diff
@@ -1083,20 +1083,11 @@ func (v2 *Handlers) GetLedgerStateDelta(ctx echo.Context, round uint64) error {
 	if err != nil {
 		return internalError(ctx, err, errFailedRetrievingStateDelta, v2.Log)
 	}
-	consensusParams, err := v2.Node.LedgerForAPI().ConsensusParams(basics.Round(round))
-	if err != nil {
-		return internalError(ctx, fmt.Errorf("unable to retrieve consensus params for round %d", round), errInternalFailure, v2.Log)
-	}
-	hdr, err := v2.Node.LedgerForAPI().BlockHdr(basics.Round(round))
-	if err != nil {
-		return internalError(ctx, fmt.Errorf("unable to retrieve block header for round %d", round), errInternalFailure, v2.Log)
-	}
-
-	response, err := stateDeltaToLedgerDelta(sDelta, consensusParams, hdr.RewardsLevel, round)
+	consensusParams := config.Consensus[sDelta.Hdr.CurrentProtocol]
+	response, err := StateDeltaToLedgerDelta(sDelta, consensusParams)
```

Comment:
@Eric-Warehime do you happen to know if this simplification is valid? I can revert if you aren't sure.

Reply:
Yeah it looks correct to me. https://github.com/algorand/go-algorand/blob/2594c329de9142ac21c9a2b30479962483614c85/data/ledger.go#L263